lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -27,16 +27,49 @@ class HttpClient
     #   :setting => value
     # }
 
-    def initialize(options={})
+    def initialize(options={})      
       @logger = options[:logger]
       # Again, in case we use DEFAULT_OPTIONS in the future, uncomment this.
       # @options = DEFAULT_OPTIONS.merge(options)
       @options = options
+
+      normalize_hosts!
```

Comment:
Generally +` on this, but we can't move much to the SafeURI constructor since it is in core. Changes there are hard to get in, and then we have cross-version issues :(